src/black/lines.py

```diff
@@ -714,7 +725,14 @@ def _maybe_empty_lines_for_class_or_def(
             else:
                 newlines = 0
         else:
-            newlines = 1 if current_line.depth else 2
+            if (
+                Preview.dummy_implementations in self.mode
+                and self.previous_line.is_stub_def
+                and (current_line.is_stub_def or current_line.is_decorator)
+            ):
+                newlines = user_hint_before
```

Comment:
This is perhaps controversial, but it allows users to group things as they want (e.g. in the test case, dummy and other are grouped separately) and solves some lookahead problems when dealing with decorators